homeassistant/components/vitrea/config_flow.py

```diff
@@ -0,0 +1,85 @@
+"""Vitrea config flow for UI setup."""
+
+from __future__ import annotations
+
+import asyncio
+import logging
+from typing import Any
+
+from vitreaclient import VitreaClient
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigFlowResult
+from homeassistant.const import CONF_HOST, CONF_PORT
+import homeassistant.helpers.config_validation as cv
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class VitreaConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Vitrea."""
+
+    VERSION = 1
+    MINOR_VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self._errors: dict[str, str] = {}
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            host = user_input[CONF_HOST]
+            port = user_input[CONF_PORT]
+
+            # Validate host
+            if not host or host.strip() == "":
+                errors["host"] = "invalid_host"
+
+            if not errors:
+                # Always use _async_test_connection for patchable connection test
+                try:
+                    await self._async_test_connection(host, port)
+                except ConnectionError:
+                    errors["base"] = "cannot_connect"
+                except Exception:  # noqa: BLE001
+                    errors["base"] = "unknown"
+                else:
+                    await self.async_set_unique_id(f"vitrea_{host}_{port}")
```

Comment:
host and port is not a suitable unique id, we need something unique for the device, not for the network config